tools/clang/lib/SPIRV/SpirvEmitter.cpp

```diff
@@ -7743,15 +7743,27 @@ SpirvInstruction *SpirvEmitter::processIntrinsicClip(const CallExpr *callExpr) {
 
 SpirvInstruction *
 SpirvEmitter::processIntrinsicClamp(const CallExpr *callExpr) {
-  // According the HLSL reference: clamp(X, Min, Max) takes 3 arguments. Each
-  // one may be int, uint, or float.
+  // According the HLSL reference: clamp(X, minVal, maxVal) takes 3 arguments.
+  // Each one may be int, uint, or float.
+
+  // The GLSL extended instruction set spec for FClamp indicates:
+  // "Result is undefined if minVal > maxVal"
+  // However, the HLSL developer may not necessarily respect this restriction.
```

Comment:
https://docs.microsoft.com/en-us/windows/desktop/direct3dhlsl/dx-graphics-hlsl-clamp

Says that min and max values are the 2nd and 3rd arguments.  It doesn't say what happens if those min / max values don't make sense.

I think this MR will cause unnecessary slowdowns.

Reply:
The main motivation for this change was to avoid hitting undefined results in Fclamp. The spec for `FClamp` in GLSL extended instruction set says

>Result is undefined if minVal > maxVal

while the dx documentation you mentioned does not say what happens if minVal > maxVal, the implementation always performs `min(max(x, minVal), maxVal)`.

I don't have a strong feeling for/against this change.